sdk/program/src/borsh0_10.rs

```diff
@@ -0,0 +1,303 @@
+#![allow(clippy::integer_arithmetic)]
+//! Utilities for the [borsh] serialization format, version 0.10.
+//!
+//! [borsh]: https://borsh.io/
+use {
+    borsh::{
+        maybestd::io::{Error, Write},
+        schema::{BorshSchema, Declaration, Definition, Fields},
+        BorshDeserialize, BorshSerialize,
+    },
+    std::collections::HashMap,
+};
+
+/// Get packed length for the given BorchSchema Declaration
+fn get_declaration_packed_len(
```

Comment:
Considering implementations for borsh 0.9 and borsh 0.10 are identical, should we use a macro to avoid copy/pasting the same code?
It would also make it easier for anyone reading the code to see that it is indeed identical.

Here is an example: https://github.com/ilya-bobyr/borsh-multi-version/blob/main/src/main.rs#L15

You can still pass the deprecated attributes as arguments to the macro.
So that function generated for the 0.9 version will be marked deprecated, while the 0.10 ones will not be.
Attributes are matched using [`meta` fragment](https://veykril.github.io/tlborm/decl-macros/minutiae/fragment-specifiers.html#meta) in the macro pattern.

Reply:
Yeah sure, done! Let me know what you think about it at [7a04bf1](https://github.com/solana-labs/solana/pull/32511/commits/7a04bf18ddd5fcf3917664f50023965ebe72b793)

Reply:
It looks good!